### PR TITLE
Add article

### PR DESCRIPTION
**Pull request by bazzel**

### CODE REVIEW

File: drafts/accept_confirm.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+```
+
+My step implementation looked like this:
+
+```ruby
+# features/step_definitions/general_steps.rb
+
+When('I delete the resource') do
+  accept_confirm { click_on 'Destroy' }
```
Comment (thnukid) [inline drafts/accept_confirm.md:33]: For reference:
https://www.rubydoc.info/github/jnicklas/capybara/Capybara%2FSession:accept_confirm

1. Execute the block, accepting a confirm.
2. Expects a block whose actions will trigger the display modal to appear.

Which Javascript Driver are you using? 

Maybe the accept_modal is not implemented in that driver.

I also remember that (most of the) drivers ignore any native dialog boxes (confirm, alert) by default.

```
# https://www.rubydoc.info/github/jnicklas/capybara/Capybara/Selenium/Driver#accept_modal-instance_method
def accept_modal(_type, **options)
  yield if block_given?
  modal = find_modal(options)

  modal.send_keys options[:with] if options[:with]

  message = modal.text
  modal.accept
  message
end
```

When using`options[:with]` and then use `:enter` for the `modal.send_keys`, that might also work.

File: drafts/accept_confirm.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+After many (!!!) hours of trying rewriting the stub, rubber ducking with colleagues, writing alternative scenarios, I decided to get rid of the confirmation dialog that was shown to the user. This way I didn't need the `accept_confirm` and... tadaaa: it worked! My test was passing.
+
+My theory was that `accept_confirm` executes in a different thread or something in which the stub is not defined (I know a theory can be proven wrong, but this one worked for me).  
+One way to work around this is, instead of using `accept_confirm`, is 'overriding' the JavaScript's `confirm` function:
+
+```ruby
+# features/step_definitions/general_steps.rb
+
+When('I delete the resource') do
+  page.evaluate_script('window.confirm = function() { return true; }')
```
Comment (thnukid) [inline drafts/accept_confirm.md:60]: So any time, the app calls the `window.confirm`, it is overwritten. So when the native confirm box is removed, the spec will still pass, without showing the confirm box. 

To make sure that the confirm box is shown, we could set a variable inside the `window.confirm` and assert if the variable was set.

File: drafts/accept_confirm.md (hunk 1 above)
Comment (bazzel) [reply drafts/accept_confirm.md:33]: The problem was not that the confirmation dialog wasn't accepted (so the `accept_modal` worked), but the `stub_request` was not recognized. But maybe that wasn't clear from the article?

File: drafts/accept_confirm.md (hunk 2 above)
Comment (bazzel) [reply drafts/accept_confirm.md:60]: Maybe that's a better solution. Will try and then add this as a remark to the article. 👍

File: drafts/accept_confirm.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# You're not stubbing, stupid!
+
+Recently, in a [Ruby on Rails](https://rubyonrails.org/) project, I was writing a [Cucumber](https://cucumber.netlify.app/docs/installation/ruby/) scenario that was deleting a resource by having the user clicking a 'Delete' button. Before the action was executed, the user had to confirm a message shown in a confirmation dialog. You may have seen this dozens of times when scaffolding a Rails application.  
+Oh and upon deleting, I also had to do a request to an external API (to be more precise: the use case was that a user unsubscribing, so I had to send a DELETE request to a [Mollie API](https://docs.mollie.com/reference/v2/subscriptions-api/cancel-subscription)).
```
Comment (ariejan) [inline drafts/accept_confirm.md:4]: was that **of** a user unsubscribing

Comment (ariejan) [reply drafts/accept_confirm.md:4]: "DELETE" => "`DELETE`"

Comment (bazzel) [reply drafts/accept_confirm.md]: ✅

File: drafts/accept_confirm.md (hunk 2 above)
Comment (bazzel) [reply drafts/accept_confirm.md:61]: I updated the article. not setting a JS variable, but checking the link/button for having a `data-confirm` attribute.

**Pull request closed without merging**